### PR TITLE
feat: Table Column Defaults

### PR DESCRIPTION
**Pull request by robbruce**
Add support for column defaults on the `snowflake_table` resource. Note there are some limitations:

- `ALTER TABLE MODIFY COLUMN` does not support adding a default; unless a sequence has been used originally as the default **and** a new sequence is being used
- `ALTER TABLE ADD COLUMN` only supports setting a column default to a constant value

## Test Plan
* [x] acceptance tests
* [x] unit tests
* [x] ran `make docs`

## References
* https://docs.snowflake.com/en/sql-reference/sql/create-table.html#optional-parameters Default section

### CODE REVIEW

**Comment by robbruce** [comment]
@alldoami - I've corrected the failed TestSequenceRead unit test; some of the other unit tests I can't run locally just to account level conflicts we already have enabled, e.g. SCIM.

**Comment by alldoami** [comment]
Looks like there is a `panic: runtime error: invalid memory address or nil pointer dereference`

**Comment by robbruce** [comment]
@alldoami - sorted. I hadn't correctly accounted for new columns being added without a default value 🤦 . Working now

```
$ TF_ACC=1 go test -v ./... -run TestAcc_Table
?   	github.com/chanzuckerberg/terraform-provider-snowflake	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/datasources	(cached) [no tests to run]
?   	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/db	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/provider	(cached) [no tests to run]
=== RUN   TestAcc_Table
=== PAUSE TestAcc_Table
=== RUN   TestAcc_TableDefaults
=== PAUSE TestAcc_TableDefaults
=== CONT  TestAcc_Table
=== CONT  TestAcc_TableDefaults
--- PASS: TestAcc_TableDefaults (11.76s)
--- PASS: TestAcc_Table (59.63s)
PASS
ok  	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/resources	59.982s
testing: warning: no tests to run
PASS
ok  	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake	(cached) [no tests to run]
?   	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/testhelpers	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/validation	(cached) [no tests to run]
?   	github.com/chanzuckerberg/terraform-provider-snowflake/pkg/version	[no test files]
```

**Comment by robbruce** [comment]
@alldoami - anything in the logs to suggest what else needs correcting?

File: pkg/resources/sequence_acceptance_test.go [hunk 1]
```diff
@@ -31,6 +32,7 @@ func TestAcc_Sequence(t *testing.T) {
 					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "name", accRename),
 					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "comment", "look at me I am a comment"),
 					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "next_value", "1"),
+					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "fully_qualified_name", fmt.Sprintf(`%v.%v.%v`, accName, accName, accName)),
```
Comment (alldoami) [inline pkg/resources/sequence_acceptance_test.go:35]:
```suggestion
					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "fully_qualified_name", fmt.Sprintf(`%v.%v.%v`, accName, accName, accRename)),
```

**Pull request merged**